### ISSUE TITLE
Some Accessibility issues

### ISSUE THREAD

**Issue by zhou-zhoukang** [t895/DNSNet#92]
Hi! Thank you for your hard work on this great open-source app. I noticed a few accessibility issues in DNSNet. 

1. Taking the example of the interface shown in the figure below, the red boxes indicate the order in which components are focused sequentially by screen readers when visually impaired users navigate through the UI. When they focus on the Switch, they will only hear its current on/off state but cannot understand what specific feature this Switch is enabling/disabling. This requires them to perform additional left/right swipes to obtain contextual information about the Switch. This may creates unnecessary cognitive load.

![Image](https://github.com/user-attachments/assets/598bce27-4d53-4a30-ac32-a51bc819e3fd)

Perhaps a more accessible approach in this case would be to set the `onCheckedChange` of the Switch to null, and instead handle the click event within the `ToggleableSetting`. This is just a suggestion based on what I’ve observed, and I hope it may be helpful.
```
ToggleableSetting(  
    modifier = Modifier.clickable { onCheckedChange() },  
    toggleableContent = {  
        MaterialSwitch(  
            onCheckedChange = null,  
        )
    }  
)
```

2. It is often considered a good practice to set an `onClickLabel` for clickable controls, which may help improve accessibility. An example of how this can be implemented is provided in the code snippet below. For instance, when considering a button, a screen reader might announce "Double tap to **activate**" before a label is added. After adding an appropriate label, it would instead say "Double tap to **enter**", which can make the button's function more intuitive and clear to users. This is just one way to enhance the user experience, and there may be other approaches as well.
```
modifiler = Modifier.semantics {  
   onClick(label = "enter") { }
}
```
Please feel absolutely no pressure — I completely understand if this takes a back seat to other priorities. Truly appreciate all the effort and hard work you've already put in!

**Comment by t895** [comment]
Thank you for bringing this up! I've actually been hoping to improve accessibility but I don't have a great grasp on what screen reader users expect.

I'll hopefully get to working on some improvements soon around these switches/submenus soon.

**Comment by t895** [comment]
I thought this would be much more work but it really was just a few lines to fix using your approach under #2
I think it's working as expected with my testing using TalkBack but I'll leave this open for comments after the next update
https://github.com/t895/DNSNet/commit/a724e653d33c8f3dcc0734f1a3082ed995c9b878
